src/Microsoft.AspNet.Http/Features/FormFeature.cs

```diff
@@ -119,7 +144,7 @@ public IFormCollection ReadForm()
                     var section = await multipartReader.ReadNextSectionAsync(cancellationToken);
                     while (section != null)
                     {
-                        var headers = new HeaderDictionary(section.Headers);
+                        var headers = section.Headers;
                         ContentDispositionHeaderValue contentDisposition;
                         ContentDispositionHeaderValue.TryParse(headers[HeaderNames.ContentDisposition], out contentDisposition);
```

Comment:
regression: this will throw if Content-Disposition is missing. You could just use section.ContentDisposition instead.